server/core/runtime/policy/conftest_executor.go

```diff
@@ -62,62 +55,61 @@ func NewConfTestExecutor(creator githubapp.ClientCreator, org string) *ConfTestE
-	var policyArgs []Arg
 	var policyNames []string
 	var failedPolicies []valid.PolicySet
+	var totalCmdOutput string
+	var policyErr error
+
 	inputFile := filepath.Join(workdir, prjCtx.GetShowResultFileName())
 	scope := prjCtx.Scope.SubScope(conftestScope)
 
 	for _, policySet := range prjCtx.PolicySets.PolicySets {
-		path, err := c.SourceResolver.Resolve(policySet)
-		// Let's not fail the whole step because of a single failure. Log and fail silently
-		if err != nil {
-			prjCtx.Log.ErrorContext(prjCtx.RequestCtx, fmt.Sprintf("Error resolving policyset %s. err: %s", policySet.Name, err.Error()))
-			continue
+		var policyArgs []Arg
+		for _, path := range policySet.Paths {
+			policyArgs = append(policyArgs, NewPolicyArg(path))
 		}
-		policyArgs = append(policyArgs, NewPolicyArg(path))
 		policyNames = append(policyNames, policySet.Name)
-	}
-
-	args := ConftestTestCommandArgs{
-		PolicyArgs: policyArgs,
-		ExtraArgs:  extraArgs,
-		InputFile:  inputFile,
-		Command:    executablePath,
-	}
-	serializedArgs, err := args.build()
-	if err != nil {
-		prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
-		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
-		return "", errors.Wrap(err, "building args")
-	}
+		args := ConftestTestCommandArgs{
+			PolicyArgs: policyArgs,
+			ExtraArgs:  extraArgs,
+			InputFile:  inputFile,
+			Command:    executablePath,
+		}
+		serializedArgs, err := args.build()
+		if err != nil {
+			prjCtx.Log.WarnContext(prjCtx.RequestCtx, "No policies have been configured")
+			scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
+			return "", errors.Wrap(err, "building args")
+		}
 
-	// TODO: run each policy set separately and use each pass/failure decision to populate failedPolicies
-	cmdOutput, policyErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
-	if policyErr != nil {
-		failedPolicies = prjCtx.PolicySets.PolicySets
+		cmdOutput, cmdErr := c.Exec.CombinedOutput(serializedArgs, envs, workdir)
+		// Continue running other policies if one fails since it might not be the only failing one
+		if cmdErr != nil {
+			policyErr = cmdErr
+			failedPolicies = append(failedPolicies, policySet)
+		}
+		totalCmdOutput = fmt.Sprintf("%s\n%s", totalCmdOutput, cmdOutput)
 	}
 
 	title := c.buildTitle(policyNames)
-	output := c.sanitizeOutput(inputFile, title+cmdOutput)
+	output := c.sanitizeOutput(inputFile, title+totalCmdOutput)
 	if prjCtx.InstallationToken == 0 {
 		prjCtx.Log.ErrorContext(prjCtx.RequestCtx, "missing installation token")
 		scope.Counter(metrics.ExecutionErrorMetric).Inc(1)
 		return output, errors.New(internalError)
 	}
 
-	failedPolicies, err = c.PolicyFilter.Filter(prjCtx.RequestCtx, prjCtx.InstallationToken, prjCtx.HeadRepo, prjCtx.Pull.Num, failedPolicies)
+	failedPolicies, err := c.PolicyFilter.Filter(prjCtx.RequestCtx, prjCtx.InstallationToken, prjCtx.HeadRepo, prjCtx.Pull.Num, failedPolicies)
```

Comment:
If I'm understanding correctly, `PolicyFIlter` will filter out policies based on approvals in the PR but the policy output will still include the output from the failing conftest runs for those filtered policies since we're not filtering the output accordingly right? Should we filter out the policy output too?

Reply:
Correct me if I'm misunderstanding your question here, but I believe this matches our current implementation of continuing to provide information on what policies did fail while allowing for the check run itself to be considered successful: https://github.com/lyft/atlantis/pull/303

Reply:
What I meant to say is for example we have policy sets A through E and for a PR with approval from policy owner of policy set A, all the policy sets are failing. The policyOutput will include the failed output from all the policy sets but since we have an approval from policy set A owner, we filter out policy set A from failed policies and only return policy sets B through E as failed policies, if I understand it correctly. Is that the expected behavior or should we keep consistency between the policy output and policy names, filtering out the policy output as well for approved policy sets?